lib/questions.rb

```diff
@@ -1,61 +1,82 @@
 # keep only the elements that start with an a
 def select_elements_starting_with_a(array)
+  array.keep_if {|array| array =~ /\A[a]/}
 end
 
 # keep only the elements that start with a vowel
 def select_elements_starting_with_vowel(array)
+  array.keep_if {|array| array =~ /\A[aeiou]/}
 end
 
 # remove instances of nil (but NOT false) from an array
 def remove_nils_from_array(array)
+  array.compact!
 end
 
 # remove instances of nil AND false from an array
 def remove_nils_and_false_from_array(array)
+  array.compact!
+  array.reject {|array| array == false}
 end
 
 # don't reverse the array, but reverse every word inside it. e.g.
 # ['dog', 'monkey'] becomes ['god', 'yeknom']
 def reverse_every_element_in_array(array)
+  array.map{|array| array.reverse}
```

Comment:
Shadowing outer local variable - `array`.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.<br>Pass `&:reverse` as an argument to `map` instead of a block.